include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -0,0 +1,552 @@
+    {
+        // increment the number of evaluated parameter combinations
+        ++stats.num_evaluated_parameter_combinations;
+
+        // take the first (and only) output BDL pair
+        const auto& output_bdl_pair = output_bdl_pairs.front();
+
+        // reset the BDL input iterator
+        bii = 0ull;
+
+        // for each input combination
+        for (auto i = 0u; i < truth_table.num_bits(); ++i, ++bii)
+        {
+            // the expected output of the layout is the i-th bit of the truth table
+            const auto expected_output = kitty::get_bit(truth_table, i);
+
+            // TODO replace with QuickExact
+            // perform an exhaustive ground state simulation
+            const auto sim_result = exhaustive_ground_state_simulation(*bii, sim_params);
+
+            // TODO is this necessary or is it guaranteed that the ground state is always the first element?
+            // find the ground state, which is the charge distribution with the lowest energy
+            const auto ground_state = std::min_element(
+                sim_result.charge_distributions.cbegin(), sim_result.charge_distributions.cend(),
+                [](const auto& lhs, const auto& rhs) { return lhs.get_system_energy() < rhs.get_system_energy(); });
+
+            // fetch the charge states of the output BDL pair
+            const auto charge_state_output_upper = ground_state->get_charge_state(output_bdl_pair.upper);
+            const auto charge_state_output_lower = ground_state->get_charge_state(output_bdl_pair.lower);
+
+            // if the output charge states are equal, the layout is not operational
+            if (charge_state_output_lower == charge_state_output_upper)
+            {
+                ++stats.num_non_operational_parameter_combinations;
+
+                return operational_domain::operational_status::NON_OPERATIONAL;
+            }
+
+            // if the expected output is 1, the expected charge states are (upper, lower) = (0, -1)
+            if (expected_output)
+            {
+                if (charge_state_output_upper != sidb_charge_state::NEUTRAL ||
+                    charge_state_output_lower != sidb_charge_state::NEGATIVE)
+                {
+                    ++stats.num_non_operational_parameter_combinations;
+
+                    return operational_domain::operational_status::NON_OPERATIONAL;
+                }
+            }
+            // if the expected output is 0, the expected charge states are (upper, lower) = (-1, 0)
+            else
+            {
+                if (charge_state_output_upper != sidb_charge_state::NEGATIVE ||
+                    charge_state_output_lower != sidb_charge_state::NEUTRAL)
+                {
+                    ++stats.num_non_operational_parameter_combinations;
+
+                    return operational_domain::operational_status::NON_OPERATIONAL;
+                }
+            }
+        }
+
```

Comment:
At this point, I am a little bit concerned since degeneracy is not covered. I think if there is degeneracy, it should be labeled as ``NON_OPERATIONAL``. Maybe you can take a look at ``energy_distribution.hpp``. It tells you not only the ground state energy, but also its degeneracy.

Reply:
Thanks, will do! :+1:

Reply:
Sorry, I was wrong. Since there can also be degeneracy which does not affect the logic output of the gate, I suggest to just check if all ground states satisfy the logic. Sorry for the wrong information above!